File: stay-local-pay-local/src/main/java/it/hackthealps/codemates/staylocalpaylocal/opendata/model/ScoreModel.java
package it.hackthealps.codemates.staylocalpaylocal.opendata.model;

import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;

import lombok.Data;

@Data
@Entity
public class ScoreModel extends BaseModel implements Comparable<ScoreModel> {

    @Enumerated(value = EnumType.STRING)
    final ScoreValue scoreValue;

    final String scoreCriteria;

    public ScoreValue getScoreValue()
    {
    	return scoreValue;
    }
    @Override
    public int compareTo(ScoreModel o) {
        return o.scoreValue.ordinal() - scoreValue.ordinal();
    }
}

File: android/app/src/main/java/com/hackthealps/paylocal/model/ScoreModel.kt
package com.hackthealps.paylocal.model

data class ScoreModel( val scoreValue: ScoreValue,  val scoreCriteria: String) : Comparable<ScoreModel> {

    override fun compareTo(o: ScoreModel): Int {
        return o.scoreValue.ordinal - this.scoreValue.ordinal
    }
}
